### PR TITLE
Google Play Store URL

### PR DESCRIPTION
**Pull request by pnemonic78**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -24,19 +24,18 @@ static def gitSha() {
...
         targetSdkVersion 33
         versionCode versionMajor * 10000 + versionMinor * 1000 + versionPatch * 100 + versionBuild
         versionName "${versionMajor}.${versionMinor}.${versionPatch}"
         resValue "string", "app_version_name", "${versionName}"
         resValue "string", "app_minor_version", "${versionMinor}"
+        resValue "string", "app_playstore_url", "market://details?id=${applicationId}"
         buildConfigField "boolean", "CAN_REQUEST_RATING", "false"
         buildConfigField "String", "BUILD_TIME", "\"${buildTime()}\""
 
         testInstrumentationRunner "org.gnucash.android.test.ui.util.GnucashAndroidTestRunner"
```
Comment (Kaligula0) [inline app/build.gradle:38]: Shouldn't it be  `org.gnucash.pocket.…` and `….GnuCashPocketTestRunner`? (Also in line 157)

File: app/build.gradle [hunk 2]
```diff
@@ -93,10 +92,9 @@ android {
 
     productFlavors {
         development {
-            applicationId 'org.gnucash.android.devel'
-            testApplicationId 'org.gnucash.android.devel.test'
-            resValue "string", "app_name", "GnuCash-devel"
+            applicationIdSuffix '.devel'
             versionName "${versionMajor}.${versionMinor}.${versionPatch}-dev${versionBuild}_r${gitSha()}"
+            resValue "string", "app_name", "GnuCash-devel"
```
Comment (Kaligula0) [inline app/build.gradle:97]: Shouldn't it be  `GnuCash Pocket-devel` or `GnuCash Pocket (dev)` or `GnuCash Pocket (alpha)` or similar?

File: app/build.gradle [hunk 3]
```diff
@@ -105,8 +103,8 @@ android {
         }
 
         beta {
-            resValue "string", "app_name", "GnuCash - beta"
             versionName "${versionMajor}.${versionMinor}.${versionPatch}-beta${versionBuild}"
+            resValue "string", "app_name", "GnuCash - beta"
```
Comment (Kaligula0) [inline app/build.gradle:107]: Shouldn't it be  `GnuCash Pocket-beta` or `GnuCash Pocket (beta)` or similar?

File: app/build.gradle (hunk 1 above)
Comment (Kaligula0) [reply app/build.gradle:38]: Also line 49 `GnuCashPocket_v${variant.versionName}.apk` ?

**Review by Kaligula0 — COMMENTED** [review]
I commented a few places where there's still the old name

File: app/build.gradle (hunk 2 above)
Comment (pnemonic78) [reply app/build.gradle:97]: It's just for development. Also "too long" names in Android don't look good.

File: app/build.gradle (hunk 1 above)
Comment (pnemonic78) [reply app/build.gradle:38]: whoever modifies his file to `org.gnucash.pocket.XXX` will get the relevant applicationId.
no need to change file names.
no need to change package names.

**Comment by djbrown** [comment]
Cross posting my comment here
https://github.com/GnuCash-Pocket/gnucash-android/pull/83#discussion_r1567106117

**Comment by Kaligula0** [comment]
> Cross posting my comment here [#83 (comment)](https://github.com/GnuCash-Pocket/gnucash-android/pull/83#discussion_r1567106117)

I can see the comment preview in popup but can't see the comment in #83 🤔

popup:
image

#83:
image

**Review by gutierri — APPROVED** [review]
As the names are already parameterized in an xml, I believe we are in agreement and approved (for me). We may change in the future.

**Comment by gutierri** [comment]
> As the names are already parameterized in an xml, I believe we are in agreement and approved (for me). We may change in the future.

you just need to fix the CI.

**Comment by Kaligula0** [comment]
> As the names are already parameterized in an xml, I believe we are in agreement and approved (for me). We may change in the future.

I pointed out what I saw, but that was all I could do as I'm quite lost in this area, so I leave it all to @gutierri and @djbrown.

**Comment by djbrown** [comment]
> I can see the comment preview in popup but can't see the comment in #83 🤔

strange 😮

I was pointing out, that I'd prefer a more general App-ID and Google Play Store page
currently it's [org.gnucash.pocket.*pnemonic*](https://play.google.com/store/apps/details?id=org.gnucash.pocket.pnemonic) -> should be just org.gnucash.pocket

---

> As the names are already parameterized in an xml, I believe we are in agreement and approved (for me). We may change in the future.

can the App-ID be easily changed in the app store?

---

> you just need to fix the CI.

I'll look into that now 😉

**Comment by djbrown** [comment]
Test Quota Mitigation implemented with #89
Some Tests still fail though.

I read up on #82 and looked through the actual code base.
There is no mentioning of [org.gnucash.pocket.pnemonic](https://play.google.com/store/apps/details?id=org.gnucash.pocket.pnemonic) (except of the README), which wasn't meant to be the official app id anyways.

I tried to create an organization developer account for GnuCash Pocket on the Google Play Store, but there are multiple problems (see [signup](https://play.google.com/console/signup) and [help](https://support.google.com/googleplay/android-developer/answer/13628312))
* we would have to register as legal (nonprofit) business and verify that via a [DUNS number](https://www.dnb.com/duns/get-a-duns.html)
* we would have to provide a phone number and email adress for google as well as users
* we would have to provide a payment method and pay a 25$ fee for the Google Play Store account

I don't think it's feasible to set up all those things in a collective community manner.
What do you think, should we use a personal account instead?
I would really like to prevent [bus factors](https://en.wikipedia.org/wiki/Bus_factor)/[MIA-scenarios](https://f-droid.org/en/docs/MIA_Process/) which incapacitate the project.
The project shouldn't depend on one single maintainer, instead maintainers should be replaceable.
Or am I too idealistic here? 😅 Is there a good middle ground for the App Store topic?

**Comment by djbrown** [comment]
closing this as with the rewritten history on this PR it is now fully contained in #102

**Pull request closed without merging**